server/public/model/config.go

```diff
@@ -3958,6 +3981,14 @@ func (s *FileSettings) isValid() *AppError {
 		return NewAppError("Config.IsValid", "model.config.is_valid.amazons3_timeout.app_error", map[string]any{"Value": *s.MaxImageDecoderConcurrency}, "", http.StatusBadRequest)
 	}
 
+	if !(*s.AmazonS3StorageClass == "" || *s.AmazonS3StorageClass == StorageClassStandard || *s.AmazonS3StorageClass == StorageClassReducedRedundancy || *s.AmazonS3StorageClass == StorageClassStandardIA || *s.AmazonS3StorageClass == StorageClassOnezoneIA || *s.AmazonS3StorageClass == StorageClassIntelligentTiering || *s.AmazonS3StorageClass == StorageClassGlacier || *s.AmazonS3StorageClass == StorageClassDeepArchive || *s.AmazonS3StorageClass == StorageClassOutposts || *s.AmazonS3StorageClass == StorageClassGlacierIR || *s.AmazonS3StorageClass == StorageClassSnow || *s.AmazonS3StorageClass == StorageClassExpressOnezone) {
```

Comment:
This can be re-written as

```go
if *s.AmazonS3StorageClass != "" && !slices.Contains([]string{..list of constants..}, s.AmazonS3StorageClass) {
  // return error
}
```